python/cudf/cudf/core/series.py

```diff
@@ -2332,6 +2338,43 @@ def apply(self, func, convert_dtype=True, args=(), **kwargs):
         1    <NA>
         2     4.5
         dtype: float64
+
+        UDFs manipulating string data are allowed, as long as
+        they neither modify strings in place nor create new strings.
+        For example, the following UDF is allowed:
+
+        >>> def f(st):
+        ...     if len(st) == 0:
+        ...             return -1
+        ...     elif st.startswith('a'):
+        ...             return 1
+        ...     elif 'example' in st:
+        ...             return 2
+        ...     else:
+        ...             return 3
+        ...
+        >>> sr = cudf.Series(['', 'abc', 'some_example'])
+        >>> sr.apply(f)
+        0   -1
+        1    1
+        2    2
+        dtype: int64
+
+        However, the following UDF is not allowed since it includes an
+            operation that requires the creation of a new string: a call to the
+            ``upper`` method. Methods that are not supported in this manner
+            will raise an ``AttributeError``.
```

Comment:
```suggestion
        However, the following UDF is not allowed since it includes an
        operation that requires the creation of a new string: a call to the
        ``upper`` method. Methods that are not supported in this manner
        will raise an ``AttributeError``.
```